Source/Atomic/Core/Context.h

```diff
@@ -228,6 +228,27 @@ class ATOMIC_API Context : public RefCounted
     void AddGlobalEventListener(GlobalEventListener* listener) { globalEventListeners_.Push(listener); }
     void RemoveGlobalEventListener(GlobalEventListener* listener) { globalEventListeners_.Erase(globalEventListeners_.Find(listener)); }
 
+    WeakPtr<Engine> engine_;

```

Comment:
What do you think about moving these raw members to```private``` and then adding method accessors?  I think would be more consistent, and will make all of these subsystems gettable off Context in C#/JS/TS and also protect them from being set...